rai/node/node.cpp

```diff
@@ -2890,6 +2922,45 @@ bool rai::active_transactions::start (MDB_txn * transaction_a, std::shared_ptr <
 	return existing != roots.end ();
 }
 
+bool rai::active_transactions::check_election_results (MDB_txn * transaction_a, std::shared_ptr <rai::block> block_a, rai::election_result & result)
+{
+	std::lock_guard <std::mutex> lock (mutex);
+	auto root (block_a->root ());
+	auto existing (roots.find (root));
+	if (existing != roots.end ())
+	{
+		existing->election->get_progress(transaction_a, result);
+		return false;
+	}
+	else
+	{
+		auto historical (election_history.get <1>().find (root));
+		if (historical != election_history.get <1>().end ())
+		{
+			result = historical->result;
+			return false;
+		}
+	}
+	return true;
+}
+
+void rai::active_transactions::add_election_history (const rai::block_hash & root, const election_result & result)
+{
+	rai::election_history history;
+	history.root = root;
+	history.result = result;
+	auto existing_history (election_history.get <1>().find (root));
```

Comment:
I'm not a boost expert, so some extra eyes here would be nice. This is supposed to be an LRU.